src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ValidationHelperServiceImpl.java

```diff
@@ -76,4 +87,31 @@ public boolean validateUserPersistedWithException(HttpStatus status) {
         }
         return true;
     }
+
+    @Override
+    public void validateUserStatusWithException(UserProfile userProfile, IdamStatus expectedStatus) {
+        if (expectedStatus != userProfile.getStatus()) {
+            auditService.persistAudit(BAD_REQUEST, API);
+            final String exceptionMsg = String.format("User is not in %s state", expectedStatus);
+            exceptionService.throwCustomRuntimeException(BADREQUEST, exceptionMsg);
+        }
+    }
+
+    @Override
+    public void validateUserLastUpdatedWithinSpecifiedTimeWithException(UserProfile userProfile, long expectedMins) {
+        if (Duration.between(userProfile.getLastUpdated(), LocalDateTime.now()).toMinutes() < expectedMins) {
+            auditService.persistAudit(HttpStatus.TOO_MANY_REQUESTS, API);
+            final String exceptionMsg = String.format(TOO_MANY_REQUESTS.getErrorMessage());
+            exceptionService.throwCustomRuntimeException(TOOMANYREQUESTS, exceptionMsg);
+        }
+    }
+
+    @Override
+    public UserProfile validateReInvitedUser(Optional<UserProfile> userProfileOpt) {
+        validateUserIsPresentWithException(userProfileOpt);
```

Comment:
why the name of the methds are used "WithException"?

Reply:
Renamed methods